Add IDL files and empty implementation to support the Cookie Store API in Service Workers

RupinMittal opened the pull request:
#### 27bb6fc299f4133ec433d4fb49a3adeaf5cf9e87
<pre>
Add IDL files and empty implementation to support the Cookie Store API in Service Workers
<a href="https://bugs.webkit.org/show_bug.cgi?id=260495">https://bugs.webkit.org/show_bug.cgi?id=260495</a>
rdar://114225599

Reviewed by Chris Dumez.

The Cookie Store API should be supported in Service Workers as well.
This is the first in a series of changes that will make this happen.

* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieListItem_attributes.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_empty.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_empty.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_multiple.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_multiple.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_single.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_single.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_delete_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_delete_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_getAll_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_getAll_multiple.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_getAll_set_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_delete_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_set_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_set_ordering.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_set_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscribe_arguments.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscribe_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscriptions_empty.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/encoding.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookieStore_subscriptions_reset.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_mismatched_subscription.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_multiple_subscriptions.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_overlapping_subscriptions.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_single_subscription.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_oncookiechange_eventhandler_single_subscription.https.any.serviceworker-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/cookie-store/CookieChangeEvent.cpp:
* Source/WebCore/Modules/cookie-store/CookieChangeEvent.h:
* Source/WebCore/Modules/cookie-store/CookieStore.h:
* Source/WebCore/Modules/cookie-store/CookieStore.idl:
* Source/WebCore/Modules/cookie-store/CookieStoreManager.cpp: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
(WebCore::CookieStoreManager::create):
(WebCore::CookieStoreManager::subscribe):
(WebCore::CookieStoreManager::getSubscriptions):
(WebCore::CookieStoreManager::unsubscribe):
* Source/WebCore/Modules/cookie-store/CookieStoreManager.h: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/CookieStoreManager.idl: Copied from Source/WebCore/Modules/cookie-store/CookieStore.idl.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.cpp: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.cpp.
(WebCore::ExtendableCookieChangeEvent::create):
(WebCore::ExtendableCookieChangeEvent::ExtendableCookieChangeEvent):
(WebCore::ExtendableCookieChangeEvent::eventInterface const):
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.h: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.idl: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEventInit.h: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEventInit.idl: Copied from Source/WebCore/Modules/cookie-store/CookieStore.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventNames.h:
* Source/WebCore/dom/EventNames.in:
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.cpp:
(WebCore::ServiceWorkerGlobalScope::cookieStore):
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.h:
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.idl:
* Source/WebCore/workers/service/ServiceWorkerRegistration.cpp:
(WebCore::ServiceWorkerRegistration::cookies):
* Source/WebCore/workers/service/ServiceWorkerRegistration.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.idl:

Canonical link: <a href="https://commits.webkit.org/267251@main">https://commits.webkit.org/267251@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/15d77f875a4a8fff2e0be44ea6064d09cc6faacf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16056 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16374 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16769 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17820 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15075 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/16244 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/19387 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16480 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17504 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16249 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16713 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13700 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18583 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13956 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14516 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21377 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/13828 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14945 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14683 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17926 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/15295 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15283 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12960 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/16256 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14517 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/4053 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3839 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18888 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/17421 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15103 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3847 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

RupinMittal commented on Source/WebCore/workers/service/ServiceWorkerGlobalScope.idl:
```diff
@@ -38,6 +38,7 @@
 
     [NewObject] Promise<undefined> skipWaiting();
 
+    attribute EventHandler oncookiechange;
```
Is this the correct way to deal with the oncookiechange part of the IDL?

```
partial interface [ServiceWorkerGlobalScope] {
     [SameObject] readonly attribute CookieStore cookieStore;
     
     attribute EventHandler oncookiechange;
```

Because there is a build failure and I suspect it has to do with this

RupinMittal replied:
I'm still looking into if this is what's causing the build failure:

<img width="1686" alt="image" src="https://github.com/WebKit/WebKit/assets/42721308/d5d64e72-acf4-4ce8-ae8c-e30d00908350">

Once I figure out the failure, I should be able to rebasline new tests that hopefully pass. But I still wanted to upload the pr as a draft to get a look at it in the meantime.

RupinMittal replied:
Ok I've managed to fix this issue. 'cookie change' had to be added to EventNames.h. I am working on getting layout tests to pass and will update the pr then.

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStore.idl:
```diff
@@ -29,8 +29,8 @@ typedef sequence<CookieListItem> CookieList;
     ActiveDOMObject,
-    EnabledBySetting=CookieStoreAPIEnabled,
-    Exposed=(Window),
+    EnabledBySetting=CookieStoreAPIEnabled&CookieStoreAPIServiceWorkerEnabled,
```
We should drop the `&CookieStoreAPIServiceWorkerEnabled` here. We want to expose this to Window even if the CookieStoreAPIServiceWorkerEnabled setting is not enabled.

cdumez commented on Source/WebCore/Modules/cookie-store/CookieStoreManager.cpp:
```diff
@@ -0,0 +1,60 @@
+
+Ref<CookieStoreManager> CookieStoreManager::create()
+{
+    return adoptRef(*new CookieStoreManager());
```
No need for `()`

cdumez commented on Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.idl:
```diff
@@ -0,0 +1,34 @@
+// https://wicg.github.io/cookie-store/#extendablecookiechangeevent
+
+[
+    Exposed=ServiceWorker
```
`EnabledBySetting=CookieStoreAPIEnabled&CookieStoreAPIServiceWorkerEnabled,`

cdumez commented on Source/WebCore/workers/service/ServiceWorkerGlobalScope.idl:
```diff
@@ -38,6 +38,7 @@
 
     [NewObject] Promise<undefined> skipWaiting();
 
+    attribute EventHandler oncookiechange;
```
`[EnabledBySetting=CookieStoreAPIEnabled&CookieStoreAPIServiceWorkerEnabled]`

cdumez commented on Source/WebCore/workers/service/ServiceWorkerRegistration.cpp:
```diff
@@ -337,6 +338,13 @@ void ServiceWorkerRegistration::getNotifications(const GetNotificationOptions& f
 
 #endif // ENABLE(NOTIFICATION_EVENT)
 
+CookieStoreManager& ServiceWorkerRegistration::cookies()
```
Aren't you missing the corresponding IDL change? Note that the IDL attribute should also be guarded with `[EnabledBySetting=CookieStoreAPIEnabled&CookieStoreAPIServiceWorkerEnabled]`

cdumez commented on Source/WebCore/workers/service/ServiceWorkerGlobalScope.h:
```diff
@@ -98,6 +99,8 @@ class ServiceWorkerGlobalScope final : public WorkerGlobalScope {
     WEBCORE_EXPORT void addConsoleMessage(MessageSource, MessageLevel, const String& message, unsigned long requestIdentifier) final;
     void enableConsoleMessageReporting() { m_consoleMessageReportingEnabled = true; }
 
+    CookieStore& cookieStore();
```
Where is the corresponding IDL change?

RupinMittal replied:
Added!

cdumez reviewed (approved):
LGTM assuming the bots are happy.

RupinMittal committed:
Add IDL files and empty implementation to support the Cookie Store API in Service Workers
https://bugs.webkit.org/show_bug.cgi?id=260495
rdar://114225599

Reviewed by Chris Dumez.

The Cookie Store API should be supported in Service Workers as well.
This is the first in a series of changes that will make this happen.

* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieListItem_attributes.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_empty.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_empty.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_multiple.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_multiple.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_single.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStoreManager_getSubscriptions_single.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_delete_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_delete_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_getAll_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_getAll_multiple.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_getAll_set_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_delete_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_set_basic.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_get_set_ordering.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_set_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_special_names.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscribe_arguments.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscribe_arguments.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/cookieStore_subscriptions_empty.https.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/encoding.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any.sharedworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/idlharness.tentative.https.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookieStore_subscriptions_reset.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_mismatched_subscription.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_multiple_subscriptions.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_overlapping_subscriptions.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_cookiechange_eventhandler_single_subscription.https.any.serviceworker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/cookie-store/serviceworker_oncookiechange_eventhandler_single_subscription.https.any.serviceworker-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Modules/cookie-store/CookieChangeEvent.cpp:
* Source/WebCore/Modules/cookie-store/CookieChangeEvent.h:
* Source/WebCore/Modules/cookie-store/CookieStore.h:
* Source/WebCore/Modules/cookie-store/CookieStore.idl:
* Source/WebCore/Modules/cookie-store/CookieStoreManager.cpp: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
(WebCore::CookieStoreManager::create):
(WebCore::CookieStoreManager::subscribe):
(WebCore::CookieStoreManager::getSubscriptions):
(WebCore::CookieStoreManager::unsubscribe):
* Source/WebCore/Modules/cookie-store/CookieStoreManager.h: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/CookieStoreManager.idl: Copied from Source/WebCore/Modules/cookie-store/CookieStore.idl.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.cpp: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.cpp.
(WebCore::ExtendableCookieChangeEvent::create):
(WebCore::ExtendableCookieChangeEvent::ExtendableCookieChangeEvent):
(WebCore::ExtendableCookieChangeEvent::eventInterface const):
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.h: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEvent.idl: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEventInit.h: Copied from Source/WebCore/Modules/cookie-store/CookieChangeEvent.h.
* Source/WebCore/Modules/cookie-store/ExtendableCookieChangeEventInit.idl: Copied from Source/WebCore/Modules/cookie-store/CookieStore.idl.
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventNames.h:
* Source/WebCore/dom/EventNames.in:
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.cpp:
(WebCore::ServiceWorkerGlobalScope::cookieStore):
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.h:
* Source/WebCore/workers/service/ServiceWorkerGlobalScope.idl:
* Source/WebCore/workers/service/ServiceWorkerRegistration.cpp:
(WebCore::ServiceWorkerRegistration::cookies):
* Source/WebCore/workers/service/ServiceWorkerRegistration.h:
* Source/WebCore/workers/service/ServiceWorkerRegistration.idl:

Canonical link: https://commits.webkit.org/267251@main
